core/store/src/trie/nibble_slice.rs

```diff
@@ -83,7 +83,7 @@ impl<'a> NibbleSlice<'a> {
     }
 
     /// Create a new nibble slice from the given HPE encoded data (e.g. output of `encoded()`).
-    pub fn from_encoded(data: &'a [u8]) -> (NibbleSlice, bool) {
+    pub fn from_encoded(data: &'a [u8]) -> (NibbleSlice<'_>, bool) {
```

Comment:
Is the lifetime parameter `'a` necessary here? Seems to me that it can be omitted.

Reply:
@bowenwang1996 It is required:

```
error[E0621]: explicit lifetime required in the type of `data`
  --> core/store/src/trie/nibble_slice.rs:87:10
   |
86 |     pub fn from_encoded(data: &[u8]) -> (NibbleSlice<'a>, bool) {
   |                               ----- help: add explicit lifetime `'a` to the type of `data`: `&'a [u8]`
87 |         (Self::new_offset(data, if data[0] & 16 == 16 { 1 } else { 2 }), data[0] & 32 == 32)
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetime `'a` required
```

Reply:
Oh nvm I didn't realize that this is under `impl<'a> NibbleSlice<'a>`.